tiledb/sm/filesystem/vfs.cc

```diff
@@ -807,6 +807,21 @@ tuple<Status, optional<std::vector<directory_entry>>> VFS::ls_with_sizes(
   return {Status::Ok(), entries};
 }
 
+template <LsCb F>
+void VFS::ls_recursive(const URI& parent, [[maybe_unused]] F cb) const {
```

Comment:
This definition should go into the header to avoid explicitly instantiation. We want to make it easy for test classes to supply entirely different functions that what we're going to use to wrap C functions for the C API.